src/browzine-summon-adapter.js

```diff
@@ -198,34 +198,225 @@ browzine.search = (function() {
+    if(title.syndeticsImageUrl) {
+      issn = getQueryVariable(title.syndeticsImageUrl, "issn");
+    }
+
+    return encodeURIComponent(issn);
+  };
+
+  function getTarget(issn) {
+    var element = $(".results-identifier:contains('" + issn + "')").closest(".results-title-row");
+    var target = element.length > 0 ? element[0] : undefined;
+
+    return target;
+  };
+
+  function getEndpoint(issn) {
+    var endpoint;
+
+    if(issn) {
+      endpoint = api + "/search?issns=" + issn.trim().replace("-", "");
+      endpoint += "&access_token=" + apiKey;
+    }
+
+    return endpoint;
+  };
+
+  function shouldEnhance(issn) {
+    return !!issn;
+  };
+
+  function addTargets(titles) {
+    titles.forEach(function(title) {
+      var issn = getIssn(title);
+      title.target = getTarget(issn);
+      title.endpoint = getEndpoint(issn);
+      title.shouldEnhance = shouldEnhance(issn);
+    });
+
+    return titles;
+  };
+
+  function getTitles(scope) {
+    var titles = [];
+
+    if(scope) {
+      if(scope.searchResultsCtrl) {
+        if(scope.searchResultsCtrl.titleData) {
+          if(scope.searchResultsCtrl.titleData.titles) {
+            titles = angular.copy(scope.searchResultsCtrl.titleData.titles);
+          }
+        }
+      }
+    }
+
+    return titles;
+  };
+
+  function getScope(searchResults) {
+    return angular.element(searchResults).scope();
+  };
+
+  function getData(response) {
+    return Array.isArray(response.data) ? response.data[0] : response.data;
+  };
+
+  function getBrowZineWebLink(data) {
+    var browzineWebLink = null;
+
+    if(data.browzineWebLink) {
+      browzineWebLink = data.browzineWebLink;
+    }
+
+    return browzineWebLink;
+  };
+
+  function getCoverImageUrl(data) {
+    var coverImageUrl = null;
+
+    if(data.coverImageUrl) {
+      coverImageUrl = data.coverImageUrl;
+    }
+
+    return coverImageUrl;
+  };
+
+  function buildTemplate(browzineWebLink) {
+    var browzineWebLinkText = "";
+    var bookIcon = "https://s3.amazonaws.com/thirdiron-assets/images/integrations/browzine_open_book_icon.png";
+
+    browzineWebLinkText = browzine.serSol360CoreJournalBrowZineWebLinkText || "View Journal in BrowZine";
+
+    var template = "<div class='browzine' style='margin: 5px 0;'>" +
+                     "<img class='browzine-book-icon' src='{bookIcon}' style='margin-top: -3px;'/> " +
+                     "<a class='browzine-web-link' href='{browzineWebLink}' target='_blank' style='font-weight: 300;'>{browzineWebLinkText}</a>" +
+                   "</div>";
+
+    template = template.replace(/{browzineWebLink}/g, browzineWebLink);
+    template = template.replace(/{browzineWebLinkText}/g, browzineWebLinkText);
+    template = template.replace(/{bookIcon}/g, bookIcon);
+
+    return template;
+  };
+
+  function adapter(searchResults) {
+    var scope = getScope(searchResults);
+    var titles = addTargets(getTitles(scope));
+
+    (function poll(titles) {
```

Comment:
Is there a reason we're wrapping this chunk within the above method in a self executable?

Reply:
That section needed clarity, Thanks Keegan! Refactored that piece to its own function and added some tests for the new function.